lsp/process_order.js

```diff
@@ -0,0 +1,423 @@
+        }
+
+        try {
+          const orderResponse = responses.lsps1CreateOrderResponse;
+          const {message} = getMessage;
+          const {fees} = getFees;
+          const orderId = makeOrderId();
+          const expiresAt = expiryDate(orderExpiryMs);
+          orderResponse.result.order_id = orderId;
+          orderResponse.result.lsp_balance_sat = message.params.lsp_balance_sat;
+          orderResponse.result.client_balance_sat = constants.maxPushAmount;
+          orderResponse.result.confirms_within_blocks = message.params.confirms_within_blocks;
+          orderResponse.result.channel_expiry_blocks = constants.channelExpiryBlocks;
+          orderResponse.result.created_at = currentDate();
+          orderResponse.result.expires_at = expiresAt;
+          orderResponse.result.announce_channel = message.params.announce_channel;
+          orderResponse.result.order_state = constants.orderStates.created;
+          orderResponse.result.payment.state = constants.paymentStates.expectPayment;
+          orderResponse.result.payment.fee_total_sat = String(fees);
+  
+          // Same as fees because push amounts are not supported
+          orderResponse.result.payment.order_total_sat = String(fees);
+
+          const secret = randomSecret();
+          const invoiceId = sha256(secret);
+  
+          const {request, id} = await createHodlInvoice({expires_at: expiresAt, id: invoiceId, lnd: args.lnd, tokens: fees});
+  
+          orderResponse.result.payment.lightning_invoice = request;
+          orderResponse.result.payment.onchain_address = null;
+          orderResponse.result.payment.min_onchain_payment_confirmations = constants.minOnchainConfs;
+          orderResponse.result.payment.onchain_payment = null;
+          orderResponse.result.channel = null;
+  
+          await sendMessageToPeer({
+            lnd: args.lnd,
+            message: encodeMessage(orderResponse),
+            public_key: args.pubkey,
+            type: args.type,
+          });
+
+          // Store the order
+          args.orders.set(orderId, JSON.stringify(orderResponse));
+
+          return {id, orderId, secret: secret.toString('hex')};
+        } catch (err) {
+          // Ignore errors
+          return;
+        }
+      }],
+
+      // Subscribe to invoice
+      subscribeToPaymentRequest: [
+        'getChainFees', 
+        'getMessage', 
+        'makeOrder',
+        ({getChainFees, getMessage, makeOrder}, cbk) => 
+      {
+        // Exit early when there is no message
+        if (!getMessage.message) {
+          return cbk();
+        }
+
+        const {id, orderId, secret} = makeOrder;
+        const sub = subscribeToInvoice({id, lnd: args.lnd});
+
+        const timeout = setTimeout(() => {
+          sub.removeAllListeners();
+
+          return cbk([503, 'TimedOutWaitingForLightningPayment']);
+        },
+        orderExpiryMs);
+
+        sub.on('invoice_updated', async invoice => {
+          try {
```

Comment:
since there are a bunch of async things that happen here, this could be abstracted out to its own file